src/android/com/megster/cordova/BluetoothSerialService.java

```diff
@@ -278,6 +176,7 @@ public synchronized void connect(BluetoothDevice device, boolean secure) {
      * @param device  The BluetoothDevice that has been connected
      */
     public synchronized void connected(BluetoothSocket socket, BluetoothDevice device, final String socketType) {
+        setState(STATE_CONNECTED); // Moved this to start of connected, as this message needs to be passed to PCMT as soon as any SPP device connects itself so that we can early subscribe for data'
```

Comment:
Moved this to start of connected, as this message needs to be passed to PCMT as soon as any SPP device connects itself so that we can early subscribe for data'